### PR TITLE
[CORE-1020] Remove parallel subjob processing and update test

### PR DESCRIPTION
**Pull request by jamesjang02**

### CODE REVIEW

File: lib/sidekiq/superworker/processor.rb [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+      def complete(item, _new_thread = true)
         Superworker.debug "JID ##{item['jid']}: Sidekiq job complete"
 
         complete_item(item)
       end
 
-      def error(worker, item, queue, exception)
-        raise "Job has nil jid: #{item}" if item['jid'].nil?
+      def error(worker, item, _queue, exception)
+        fail "Job has nil jid: #{item}" if item['jid'].nil?
```
Comment (rahimjad) [inline lib/sidekiq/superworker/processor.rb:12]: this is minor, any reason for `fail` over `raise`?

File: lib/sidekiq/superworker/server/middleware.rb [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-        def initialize(options=nil)
+        def initialize(_options = nil)
           @processor = Sidekiq::Superworker::Processor.new
         end
 
         def call(worker, item, queue)
           begin
             return_value = yield
-          rescue Exception => exception
+          rescue => exception
```
Comment (rahimjad) [inline lib/sidekiq/superworker/server/middleware.rb:14]: 👍

File: lib/sidekiq/superworker/subjob.rb [hunk 3]
```diff
@@ -1,18 +1,21 @@
+# frozen_string_literal: true
 module Sidekiq
   module Superworker
     class Subjob
       include ActiveModel::Validations
       include ActiveModel::Naming
 
-      ATTRIBUTES = [:subjob_id, :superjob_id, :parent_id, :next_id, :children_ids, :subworker_class,
-        :superworker_class, :arg_keys, :arg_values, :status, :descendants_are_complete, :meta]
+      ATTRIBUTES = %i[
```
Comment (rahimjad) [inline lib/sidekiq/superworker/subjob.rb:8]: 👍

**Review by rahimjad — APPROVED** [review]

File: lib/sidekiq/superworker/processor.rb (hunk 1 above)
Comment (jamesjang02) [reply lib/sidekiq/superworker/processor.rb:12]: styling where raise is generally expected when you rescue vs. fail when you need to.

**Review by lamp — APPROVED** [review]

**Comment by lamp** [comment]
Ready to merge whenever @jamesjang02 feels inclined.

**Pull request merged**